lib/Display/Display.cpp

```diff
@@ -57,57 +53,66 @@ void Display::init() {
+
 void Display::_setup() {
-  printf("    Setup 'ILI9341' with: SPI_CLK=%d, SPI_MOSI=%d, SPI_MISO=%d, SPI_CS_TFT=%d.\n", SPI_CLK, SPI_MOSI, SPI_MISO, SPI_CS_TFT);
-  status = DISPLAY_STATUS::CONSOLE;
-  int height = 0;
-  int width = 0;
+  printf("    Setup 'ILI9341' for '%s' with: SPI_CLK=%d, SPI_MOSI=%d, SPI_MISO=%d, SPI_CS_TFT=%d.\n", getName().c_str(), SPI_CLK, SPI_MOSI,
+         SPI_MISO, SPI_CS_TFT);
+  height = 0;
+  width = 0;
   try {
-    xSemaphoreTake(spiBus.mutex, portMAX_DELAY);
-    tft = Adafruit_ILI9341(SPI_CS_TFT, SPI_DC, SPI_MOSI, SPI_CLK, SPI_RST, SPI_MISO);
+    uint8_t rdmode = 0;
+    uint8_t rdmadctl = 0;
+    uint8_t rdpixfmt = 0;
+    uint8_t rdimgfmt = 0;
+    uint8_t rdselfdiag = 0;
+
+    xSemaphoreTakeT(spiBus.mutex);
     tft.begin();
+    tft.setRotation(1);
     height = tft.height();
     width = tft.width();
-    tft.setRotation(1);
+    rdmode = tft.readcommand8(ILI9341_RDMODE);
+    rdmadctl = tft.readcommand8(ILI9341_RDMADCTL);
+    rdpixfmt = tft.readcommand8(ILI9341_RDPIXFMT);
+    rdimgfmt = tft.readcommand8(ILI9341_RDIMGFMT);
+    rdselfdiag = tft.readcommand8(ILI9341_RDSELFDIAG);
+    tft.setCursor(0, 0);
     tft.setTextSize(1);
-    tft.fillScreen(ILI9341_WHITE);
+    tft.fillScreen(bgColor);
     tft.setTextColor(ILI9341_BLUE);
     tft.setScrollMargins(0, height);
-    uint8_t rdmode = tft.readcommand8(ILI9341_RDMODE);
-    uint8_t rdmadctl = tft.readcommand8(ILI9341_RDMADCTL);
-    uint8_t rdpixfmt = tft.readcommand8(ILI9341_RDPIXFMT);
-    uint8_t rdimgfmt = tft.readcommand8(ILI9341_RDIMGFMT);
-    uint8_t rdselfdiag = tft.readcommand8(ILI9341_RDSELFDIAG);
-    tft.printf("Screen ILI9341 inited (%dx%d)-->%s\n", height, width, DISPLAY_STATUS_str[(int)status]);
+    tft.printf("%s inited (%dx%d)-->%s\n", getName().c_str(), height, width, DISPLAY_STATUS_str[(int)carState.displayStatus]);
     xSemaphoreGive(spiBus.mutex);
-    printf("    ILI9341_RDMODE:     0x%x\n", rdmode);
+
     printf("    ILI9341_RDMADCTL:   0x%x\n", rdmadctl);
     printf("    ILI9341_RDPIXFMT:   0x%x\n", rdpixfmt);
     printf("    ILI9341_RDIMGFMT:   0x%x\n", rdimgfmt);
     printf("    ILI9341_RDSELFDIAG: 0x%x\n", rdselfdiag);
+    printf("    ILI9341_RDMODE:     0x%x\n", rdmode);
   } catch (__exception ex) {
     printf("[x] Display: Unable to initialize screen 'ILI9341'.\n");
     xSemaphoreGive(spiBus.mutex);
     throw ex;
   }
   lifeSignX = width - lifeSignRadius - 6;
   lifeSignY = height - lifeSignRadius - 6;
-  printf("[v] Display inited: screen 'ILI9341' with %d x %d. Status: %s\n", height, width, DISPLAY_STATUS_str[(int)status]);
+  printf("[v] '%s' Display inited: screen 'ILI9341' with %d x %d. Status: %s\n", getName().c_str(), height, width,
+         DISPLAY_STATUS_str[(int)carState.displayStatus]);
 }
 
 void Display::clear_screen(int bgColor) {
-  xSemaphoreTake(spiBus.mutex, portMAX_DELAY);
+  xSemaphoreTakeT(spiBus.mutex);
```

Comment:
What is the difference between xSemaphoreTakeT and xSemaphoreTake?

Reply:
Writes a text to serial console and throws error in case of not successful Take